### PR TITLE
Add support for themes

### PR DESCRIPTION
**Pull request by psiinon**
It will conflict a little with #2858 and #2863 :/

Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

**Comment by psiinon** [comment]
Just added help ;)
The dialog now looks like:
Screenshot 2021-04-27 at 09 41 15

And the new Dark report looks like:
Screenshot_2021-04-27 ZAP Scanning Report

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by ricekot — APPROVED** [review]

File: addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportDialog.java [hunk 1]
```diff
@@ -297,12 +306,27 @@ private JScrollPane getSectionsScrollPane() {
...
         sectionPanel.setLayout(new BoxLayout(sectionPanel, BoxLayout.Y_AXIS));
 
         Template template = extension.getTemplateByDisplayName(getStringValue(FIELD_TEMPLATE));
         if (template != null) {
+            JComboBox<String> themeCombo = ((JComboBox<String>) this.getField(FIELD_THEME));
+            themeCombo.removeAllItems();
+            for (String theme : template.getThemeNames()) {
+                themeCombo.addItem(theme);
+            }
+            if (template.getThemeNames().size() > 0) {
```
Comment (kingthorin) [inline addOns/reports/src/main/java/org/zaproxy/addon/reports/ReportDialog.java:321]: Not empty?

**Review by kingthorin — APPROVED** [review]

**Comment by kingthorin** [comment]
Oops I guess I took so long.....darn work meetings.

**Pull request merged**